crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -87,24 +88,77 @@ impl SimpleFaucet {
         })
     }
 
-    async fn select_coins(&self, number_of_coins: usize) -> Vec<ObjectID> {
+    async fn select_coins(
+        &self,
+        number_of_coins: usize,
+        transfer_amount: u64,
+    ) -> anyhow::Result<Vec<ObjectID>> {
         assert!(number_of_coins > 0);
         // If the gas candidate queue is exhausted, the request will be
         // suspended indefinitely until a producer puts in more candidate
         // gas objects. At the same time, other requests will be blocked by the
         // lock acquisition as well.
         let mut consumer = self.consumer.lock().await;
         let mut coins = Vec::with_capacity(number_of_coins);
+        let mut coins_to_immediately_recycle = Vec::new();
```

Comment:
Curious: does this eventually compact the coins to minimize dust?

Reply:
no, faucet needs more love to get that smart :)